src/main.rs

```diff
@@ -29,42 +29,22 @@ async fn main() {
         Err(_err) => return,
     };
 
-    let provision_with_password = imds::get_provision_with_password(&imds_body);
-    let disable_authentication = match provision_with_password {
+    if match imds::get_provision_with_password(&imds_body) {
```

Comment:
The user can set disablePasswordAuthentication to true but provide an ssh-key and not password. We shouldn't fail provisioning in this case. 
Provisioning should only fail if we can detect that there's a password to be provisioned. The only way to know for certain that password is provided is by inspecting the ovf-env.xml

Reply:
Thanks. This part was done by splitting the username/password part into a separate function `get_username`.